keps/sig-network/20200313-service-fallback-selector.md

```diff
@@ -0,0 +1,299 @@
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+This KEP proposes an extension on the existing Service API to specify a fallback mechanism for the existing selector field,
+so when the Service's selector doesn't match any ready pods, an alternate set of pods can be selected instead.
+
+## Motivation
+
+The current behavior implemented by `kube-proxy` when a `ClusterIP` Service has no ready endpoints is to reject all connections.
+This results in a poor user experience, especially for services pointing at an HTTP application where the user's browser displays an unhelpful "connection refused" error message or other generic "service unavailable" page. This proposal enables cluster users to customize the behavior that occurs when a Service has no ready pods.
+
+### Goals
+
+- Define a flexible mechanism by which Kubernetes users can adjust the behavior for when no ready pods are matched by a Service's selector.
+- The mechanism should have the same effect as adjusting the service's `spec.selector` field, even if no such change is actually made.
+
+### Non-Goals
+
+- Extend the capabilities of Service beyond Layer 4 (e.g. making `kube-proxy` HTTP-aware).
+- Provide cross-namespace routing capabilities (e.g. for a single "server down" page shared by multiple namespaces)
+- Change the behavior/API of topology-aware service routing or EndpointSlice.
+- Support partial fallback, or balancing between two sets of selectors.
+
+## Proposal
+
+### API Changes
+
+*TODO*
```

Comment:
I assume the proposal here is literally a second selector?  Is there more to it than that?